src/Centreon/Domain/PlatformTopology/PlatformTopology.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+/*
+ * Copyright 2005 - 2020 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\PlatformTopology;
+
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class designed to retrieve servers to be added using the wizard
+ *
+ */
+class PlatformTopology
+{
+    /**
+     * Available server types
+     */
+    private const SERVER_TYPE_CENTRAL = 0;
+    private const SERVER_TYPE_POLLER = 1;
+    private const SERVER_TYPE_REMOTE = 2;
+    private const SERVER_TYPE_MAP = 3;
+    private const SERVER_TYPE_MBI = 4;
+
+    /**
+     * Used to dynamically concatenate the thrown error when checking IP validity
+     */
+    private const KIND_SERVER = 'platform';
+    private const KIND_PARENT = 'parent platform';
+
+    /**
+     * @var string Server name
+     */
+    private $serverName;
+    /**
+     * @var int Server type
+     */
+    private $serverType;
+    /**
+     * @var string Server IP address
+     */
+    private $serverAddress;
+    /**
+     * @var string Server parent IP
+     */
+    private $serverParentAddress;
+    /**
+     * @var int Server parent id
+     */
+    private $serverParentId;
+
+    /**
+     * @return int
+     */
+    public function getServerType(): int
+    {
+        return $this->serverType;
+    }
+
+    /**
+     * @param int $serverType server type
+     *      SERVER_TYPE_CENTRAL = 0
+     *      SERVER_TYPE_POLLER  = 1
+     *      SERVER_TYPE_REMOTE  = 2
+     *      SERVER_TYPE_MAP     = 3
+     *      SERVER_TYPE_MBI     = 4
+     *
+     * @return $this
+     * @throws PlatformTopologyException
+     */
+    public function setServerType(int $serverType): self
+    {
+        // The API should not be used to add a Central to another Central
+        if (self::SERVER_TYPE_CENTRAL === $serverType) {
+            throw new PlatformTopologyException(
+                sprintf(
+                    _("You cannot link the Central '%s'@'%s' to another Central"),
```

Comment:
The error message will not be clear if one of the two arguments is not filled in when the error occurs.

Reply:
Each of these values are check and sanitized before. If one of these do not comply with te requirements, it will throw an error and you won't be able to this line